lib/index.js

```diff
@@ -1723,32 +1740,23 @@ exposeMethod('pluck');
  */
 
 Stream.prototype.pickBy = function (f) {
-
-    return this.consume(function (err, x, push, next) {
+    return this.map(objectOnly(function (x) {
         var out = {};
-        if (err) {
-            push(err);
-            next();
-        }
-        else if (x === nil) {
-            push(err, x);
-        }
-        else if (_.isObject(x)) {
-            for (var k in x) {
-                if (f(k, x[k])) {
-                    out[k] = x[k];
-                }
+        var props = [];  // cache of already seen properties
+        var obj = x;  // variable used to transverse prototype chain
+
+        // function created here so we don't create one per loop
+        var testAndAdd = function (prop) {
+            if (props.indexOf(prop) === -1 && f(prop, x[prop])) {
+                out[prop] = x[prop];
             }
-            push(null, out);
-            next();
-        }
-        else {
-            push(new Error(
-                'Expected Object, got ' + (typeof x)
-            ));
-            next();
-        }
-    });
+        };
+        do {
+            Object.getOwnPropertyNames(obj).forEach(testAndAdd);
+            obj = Object.getPrototypeOf(obj);
```

Comment:
@brentpayne We have ES5 detection and the inherited keys function around the [streamifyAll implementation](https://github.com/caolan/highland/blob/master/lib/index.js#L4012). We should fallback to a `for...in` for ES5 (and mention this in the docs).